notebooks_community/FuRBO/FuRBO.ipynb

```diff
@@ -0,0 +1,1011 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 173,
+   "id": "020338e2-9eaf-49cf-9e5c-fd00e1a46835",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gpytorch\n",
+    "import numpy as np\n",
+    "\n",
+    "from botorch.fit import fit_gpytorch_mll\n",
+    "from botorch.models import SingleTaskGP\n",
+    "from botorch.models.transforms.outcome import Standardize\n",
+    "\n",
+    "from gpytorch.constraints import Interval\n",
+    "from gpytorch.kernels import MaternKernel, ScaleKernel\n",
+    "from gpytorch.likelihoods import GaussianLikelihood\n",
+    "from gpytorch.mlls import ExactMarginalLogLikelihood\n",
+    "\n",
+    "from scipy.stats import invgauss\n",
+    "from scipy.stats import ecdf\n",
+    "\n",
+    "from torch import Tensor\n",
+    "\n",
+    "def get_best_index_for_batch(n_tr, Y: Tensor, C: Tensor):\n",
+    "    \"\"\"Return the index for the best point. One for each trust region.\n",
+    "    For reference, see https://botorch.org/docs/tutorials/scalable_constrained_bo/\"\"\"\n",
+    "    is_feas = (C <= 0).all(dim=-1)\n",
+    "    if is_feas.any():  # Choose best feasible candidate\n",
+    "        score = Y.clone()\n",
+    "        score[~is_feas] = -float(\"inf\")\n",
+    "        return torch.topk(score.reshape(-1), k=n_tr).indices\n",
+    "    return torch.topk(C.clamp(min=0).sum(dim=-1), k=n_tr, largest=False).indices # Return smallest violation\n",
+    "\n",
+    "def get_fitted_model(X,\n",
+    "                     Y,\n",
+    "                     dim,\n",
+    "                     max_cholesky_size):\n",
+    "    '''Function to fit a GPR to a given set of data.\n",
+    "    For reference, see https://botorch.org/docs/tutorials/scalable_constrained_bo/'''\n",
+    "    likelihood = GaussianLikelihood(noise_constraint=Interval(1e-8, 1e-3))\n",
+    "    covar_module = ScaleKernel(  # Use the same lengthscale prior as in the TuRBO paper\n",
+    "        MaternKernel(nu=2.5, ard_num_dims=dim, lengthscale_constraint=Interval(0.005, 4.0))\n",
+    "    )\n",
+    "    model = SingleTaskGP(\n",
+    "        X,\n",
+    "        Y,\n",
+    "        covar_module=covar_module,\n",
+    "        likelihood=likelihood,\n",
+    "        outcome_transform=Standardize(m=1),\n",
+    "    )\n",
+    "    mll = ExactMarginalLogLikelihood(model.likelihood, model)\n",
+    "\n",
+    "    with gpytorch.settings.max_cholesky_size(max_cholesky_size):\n",
+    "        fit_gpytorch_mll(mll, \n",
+    "                         optimizer_kwargs={'method': 'L-BFGS-B'})\n",
+    "\n",
+    "    return model\n",
+    "\n",
+    "from botorch.models.model_list_gp_regression import ModelListGP\n",
+    "from torch.quasirandom import SobolEngine\n",
+    "\n",
+    "class Furbo_state():\n",
```

Comment:
Please use pythonic class naming (i.e. `FurboState`). Ideally, use type hints too, so that users understan what is going on.